js/theme.js

```diff
@@ -64,6 +64,18 @@ document.addEventListener('DOMContentLoaded', () => {
       document.dispatchEvent(new CustomEvent('theme:toggle-theme-mode', event));
     });
 
+  // Sphinx compatibility.
+  // Prior to Sphinx 7.2, URL_ROOT was part of DOCUMENTATION_OPTIONS.
+  let URL_ROOT = "";
+  // Sphinx >= 7.2
+  if (document.documentElement.dataset.content_root) {
+    URL_ROOT = document.documentElement.dataset.content_root;
+  }
+  // Sphinx < 7.2
+  else if (typeof DOCUMENTATION_OPTIONS.URL_ROOT !== 'undefined') {
+    URL_ROOT = DOCUMENTATION_OPTIONS.URL_ROOT;
+  }
```

Comment:
And then the reason why we need to make JS changes, is because this is a custom "autocomplete" implementation using the `autocompleter` JS package. This comes from the Typo3 docs theme that we forked (you can still see remnants of this, e.g. `window.T3Docs`).

Sphinx does not provide autocomplete – the default behaviour is that submitting the search box will direct the browser to the `search.html` page with the `q` parameter added, e.g. https://sphinx-wagtail-theme.readthedocs.io/en/v6.3.0/search.html?q=contributing. This page will then work on the client-side using Sphinx's search index, powered by the `searchtools.js` that [comes from the `basic` theme](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/themes/basic/static/searchtools.js) and is included in https://github.com/wagtail/sphinx-wagtail-theme/blob/db925d85c2c260a55817da5048e30a667e64c0b7/sphinx_wagtail_theme/layout.html#L184

In fact this is also referred a few lines below:

https://github.com/wagtail/sphinx-wagtail-theme/blob/db925d85c2c260a55817da5048e30a667e64c0b7/js/theme.js#L119-L120

The `searchtools.js` API is undocumented, hence why we need to keep track of the changes in our theme's JS file. See also https://groups.google.com/g/sphinx-users/c/qtLQXDDHbYo

Typo3's theme never runs into this issue because [they pin their Sphinx version to <5](https://github.com/TYPO3-Documentation/sphinx_typo3_theme/blob/5c80df89e833eddff14458c2c3fe56a8a602fee9/requirements_dev.txt#L1) 🫠